test/e2e/storage/non_csi_volumes.go

```diff
@@ -0,0 +1,1567 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	clientset "k8s.io/client-go/kubernetes"
+	kubeletapis "k8s.io/kubernetes/pkg/kubelet/apis"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	vspheretest "k8s.io/kubernetes/test/e2e/storage/vsphere"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+var testDrivers = map[string]func(f *framework.Framework, config framework.VolumeTestConfig, node v1.Node) testDriver{
+	"NFS":                        initNFSDriver,
+	"GlusterFS":                  initGlusterFSDriver,
+	"iSCSI [Feature:Volumes]":    initISCSIDriver,
+	"Ceph-RBD [Feature:Volumes]": initRbdDriver,
+	"CephFS [Feature:Volumes]":   initCephFSDriver,
+	"hostpath":                   initHostpathDriver,
+	"hostpathSymlink":            initHostpathSymlinkDriver,
+	"emptydir":                   initEmptydirDriver,
+	"cinder":                     initCinderDriver,
+	"gce":                        initGceDriver,
+	"vSphere":                    initVSphereDriver,
+	"azure":                      initAzureDriver,
+	"aws":                        initAwsDriver,
+}
+
+var testCases = map[string]func() testCase{
+	"volumes":                          initVolumesTestCase,
+	"volumeIO":                         initVolumeIOTestCase,
+	"volumeMode [Feature:BlockVolume]": initVolumeModeTestCase,
+	"subPath":                          initSubPathTestCase,
+}
+
+var _ = utils.SIGDescribe("Non-CSI", func() {
+	f := framework.NewDefaultFramework("volumes")
+
+	var (
+		cs     clientset.Interface
+		ns     *v1.Namespace
+		node   v1.Node
+		config framework.VolumeTestConfig
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		nodes := framework.GetReadySchedulableNodesOrDie(cs)
+		node = nodes.Items[rand.Intn(len(nodes.Items))]
+		config = framework.VolumeTestConfig{
+			Namespace:      ns.Name,
+			Prefix:         "volume",
+			ClientNodeName: node.Name,
```

Comment:
Removed it from there and only put it in hostpath and hostpathSymlink's ```createVolume```,
because hostpath is not persistent across node and it should always be scheduled on the same node.

We might need to add ```isPersistentAcrossNode``` to ```driverInfo``` and skip topology related tests for drivers like hostpath in the future.